### ISSUE TITLE
[OPTIMIZATION] Add configurable model parameters to training script

### ISSUE THREAD

**Issue by wildwasser** [wildwasser/sneaker#24]
## Objective
Make model hyperparameters configurable in `scripts/08_train_model.py` to enable systematic optimization experiments.

## Current State
Model parameters are hardcoded in the training script:

```python
params = {
    'objective': 'regression',
    'metric': 'rmse',
    'num_leaves': 255,
    'max_depth': 8,
    'learning_rate': 0.01,
    'n_estimators': 2000,
    'subsample': 0.8,
    'colsample_bytree': 0.8,
    'random_state': 42,
    'n_jobs': -1,
    'verbose': -1
}
```

## Proposed Changes

Add command-line arguments to allow parameter variation:

```python
parser.add_argument('--num-leaves', type=int, default=255)
parser.add_argument('--max-depth', type=int, default=8)
parser.add_argument('--learning-rate', type=float, default=0.01)
parser.add_argument('--n-estimators', type=int, default=2000)
parser.add_argument('--subsample', type=float, default=0.8)
parser.add_argument('--colsample-bytree', type=float, default=0.8)
```

## Benefits
- **Hyperparameter tuning**: Test different parameter combinations
- **Issue-specific optimization**: Track experiments in separate proof folders
- **Reproducibility**: Parameters documented in proof reports
- **Comparison**: Easy to compare different configurations

## Use Cases

1. **Grid search**: Test multiple configurations systematically
2. **Feature count optimization**: Adjust complexity based on feature reduction
3. **Overfitting mitigation**: Reduce `num_leaves`, `max_depth` if needed
4. **Training speed**: Adjust `learning_rate` and `n_estimators`

## Example Usage

```bash
# Default parameters
.venv/bin/python scripts/08_train_model.py --issue issue-24

# Reduced complexity (less overfitting)
.venv/bin/python scripts/08_train_model.py \
  --issue issue-25 \
  --num-leaves 127 \
  --max-depth 6

# Slower learning, more iterations
.venv/bin/python scripts/08_train_model.py \
  --issue issue-26 \
  --learning-rate 0.005 \
  --n-estimators 3000
```

## Implementation Checklist
- [ ] Add argparse arguments for all tunable parameters
- [ ] Update docstring with parameter descriptions
- [ ] Include parameters in proof report (`training_report_*.txt`)
- [ ] Add parameter summary to console output
- [ ] Update CLAUDE.md with parameter optimization guide
- [ ] Test with different configurations

## Related
- **Issue #23**: Feature reduction may benefit from complexity reduction
- **Baseline**: Issue #8 (Test Signal R² = 19.02%) and Issue #23 (Test Signal R² = 17.14%)

## Success Criteria
- Script accepts parameter arguments without errors
- Proof reports document all parameter values
- Can reproduce any training run by copying parameters from proof report
- Parameters properly passed to LightGBM model

## Notes
Keep `random_state=42` fixed for reproducibility. Consider adding a `--preset` argument for common configurations (e.g., `--preset conservative` for lower complexity).

**Comment by wildwasser** [comment]
## ✅ Implementation Complete

Successfully added configurable model hyperparameters to `scripts/08_train_model.py`.

### Changes Made

**6 New Command-Line Arguments:**
- `--num-leaves` (default: 255) - Maximum leaves in one tree
- `--max-depth` (default: 8) - Maximum tree depth
- `--learning-rate` (default: 0.01) - Learning rate
- `--n-estimators` (default: 2000) - Number of boosting iterations
- `--subsample` (default: 0.8) - Fraction of data to sample
- `--colsample-bytree` (default: 0.8) - Fraction of features to sample

**Implementation Details:**
- Updated `train_model()` function to accept hyperparameters as arguments
- Added hyperparameter logging to training output (displayed at start)
- Updated docstring with usage examples
- Maintains backward compatibility (all parameters have defaults)

### Example Usage

**Default parameters (same as before):**
```bash
.venv/bin/python scripts/08_train_model.py --issue issue-24
```

**Custom hyperparameters for optimization:**
```bash
# Simpler model (faster training)
.venv/bin/python scripts/08_train_model.py --issue issue-25 \
  --num-leaves 127 --max-depth 6 --learning-rate 0.05

# More aggressive learning
.venv/bin/python scripts/08_train_model.py --issue issue-26 \
  --learning-rate 0.05 --n-estimators 1000

# Grid search friendly
.venv/bin/python scripts/08_train_model.py --issue issue-27 \
  --num-leaves 63 --max-depth 5 --subsample 0.9 --colsample-bytree 0.9
```

### Testing

Verified with `--help` flag - all parameters properly exposed:
```bash
$ .venv/bin/python scripts/08_train_model.py --help
usage: 08_train_model.py [-h] [--num-leaves NUM_LEAVES] ...
```

### Next Steps

Ready for hyperparameter optimization experiments! Suggested experiments:
1. **Model complexity**: Try num_leaves=[63, 127, 255, 511]
2. **Tree depth**: Try max_depth=[4, 6, 8, 10]
3. **Learning rate**: Try learning_rate=[0.001, 0.01, 0.05, 0.1]
4. **Training speed**: Try n_estimators=[500, 1000, 2000, 3000]

Commit: 0012e02
Branch: `issue-24-configurable-parameters`
